selenium/src/test/java/au/org/emii/portal/tests/BaseTest.java

```diff
@@ -0,0 +1,128 @@
+import java.net.URL;
+import java.util.Properties;
+
+public class BaseTest {
+    public static final int invocationCount = 1;
+    public static final int threadPoolSize = 1;
+    public static String BROWSER_STACK_USERNAME;
+    public static String BROWSER_STACK_AUTOMATE_KEY;
+    public static String BROWSER_STACK_URL;
+    public static String BROWSER_STACK_LOCAL_URL;
+    public static String BROWSER_STACK_DEBUG;
+    public static String BROWSER_STACK_LOCAL;
+    public static String BROWSER_STACK_VIDEO;
+    public static String BROWSER_STACK_BUILD;
+    public static String AODN_PORTAL_HOME_PAGE;
+    public static String AODN_PORTAL_SEARCH_PAGE;
+    public static Properties properties = new Properties();
+    private static Logger log = Logger.getLogger(BaseTest.class.getName());
+
+    private WebDriver driver;
+
+    @Parameters({"browser", "browser_version", "os", "os_version", "device", "platform", "resolution"})
+    @BeforeClass
+    public void setup(String browser, String browser_version, String os, String os_version, String device, String platform, String resolution) throws MalformedURLException, InterruptedException {
+
+        BROWSER_STACK_AUTOMATE_KEY = System.getProperty("browserstack.automateKey");
+        BROWSER_STACK_USERNAME = System.getProperty("browserstack.username");
+        BROWSER_STACK_DEBUG = System.getProperty("browserstack.debug");
+        BROWSER_STACK_VIDEO = System.getProperty("browserstack.video");
+        BROWSER_STACK_BUILD = System.getProperty("build");
+        BROWSER_STACK_LOCAL = System.getProperty("browserstack.local");
+
+        log.debug("BROWSER_STACK_AUTOMATE_KEY: " + BROWSER_STACK_AUTOMATE_KEY);
+        log.debug("BROWSER_STACK_USERNAME: " + BROWSER_STACK_USERNAME);
+        log.debug("BROWSER_STACK_DEBUG: " + BROWSER_STACK_DEBUG);
+        log.debug("BROWSER_STACK_VIDEO: " + BROWSER_STACK_VIDEO);
+        log.debug("BROWSER_STACK_BUILD: " + BROWSER_STACK_BUILD);
+        log.debug("BROWSER_STACK_LOCAL: " + BROWSER_STACK_LOCAL);
+
+        BROWSER_STACK_URL = "https://" + BROWSER_STACK_USERNAME + ":" + BROWSER_STACK_AUTOMATE_KEY + "@hub-cloud.browserstack.com/wd/hub";
+        BROWSER_STACK_LOCAL_URL = "https://" + BROWSER_STACK_USERNAME + ":" + BROWSER_STACK_AUTOMATE_KEY + "@hub.browserstack.com/wd/hub";
+
+        AODN_PORTAL_HOME_PAGE = System.getProperty("aodnPortal");
+        AODN_PORTAL_SEARCH_PAGE = AODN_PORTAL_HOME_PAGE + System.getProperty("aodnPortalSearch");
+
+        DesiredCapabilities capability = getDesiredCapability(browser, browser_version, os, os_version, device, platform, resolution);
+
+        if (BROWSER_STACK_LOCAL.equals("true")) {
+            driver = new RemoteWebDriver(new URL(BROWSER_STACK_LOCAL_URL), capability);
+        } else {
+            driver = new RemoteWebDriver(new URL(BROWSER_STACK_URL), capability);
+        }
+
+        driver.get(AODN_PORTAL_HOME_PAGE);
+        wait(5);
+    }
+
+    @org.testng.annotations.AfterClass
```

Comment:
As above, I can see now that this is as per browserstack/testng examples for parameterized tests so this looks fine.